### ISSUE TITLE
RAG0009: Embedding Model Finetuning (2)

### ISSUE THREAD

**Issue by tenzin3** [OpenPecha/RAG_chatbot_interface#6]
## Description
We will finetune the 'Alibaba-NLP/gte-large-en-v1.5' model using the QA pairs generated in RAG0008 and select another model from the MTEB Leaderboard (https://huggingface.co/spaces/mteb/leaderboard) as benchmark. The idea is to teach the model domain specific knowledge it might not have learnt during pretraining. An additional hyperparameter tuning can also be conducted as we have access to an example code from another project. 

## Expected Output
The output is a finetuned embedding model that will be used for vectorizing the chunks as well as used during user query transformation.

## Implementation Steps
- [x] finetune Alibaba-NLP/gte-large-en-v1.5
- [x] modify script to store chunks of different books in one database
- [x] evaluate the finetuned model

**Comment by tenzin3** [comment]
## Fine-tuning Alibaba-NLP/gte-large-en-v1.5 embedding model

Data: Content from My land and my people. Refer to [here](https://github.com/OpenPecha/rag_prep_tool/issues/11)

![Image](https://github.com/OpenPecha/RAG_chatbot_interface/assets/52460417/b9776551-6f1e-4441-ac99-badcb397202c)

Above is result of one epoch. Tuning more epoch is not decreasing the loss or increasing the accuracy.

[uploaded here](https://huggingface.co/openpecha/Finetuned_Alibaba_Large)

**Comment by tenzin3** [comment]
After running for 393 questions and sources.

Hit Ratio mean: 
**Alibaba-NLP/gte-large-en-v1.5:**              0.2926208651399491 
**openpecha/Finetuned_Alibaba_Large:**    0.37659033078880405 (fine-tuned)

**Comment by tenzin3** [comment]
The embedding model is fine-tuned based on questions generated from the following

- My land and my people
- Freedom in Exile
